Tests/SwiftDiagnosticsTest/FixItTests.swift

```diff
@@ -0,0 +1,42 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftParser
+import SwiftParserDiagnostics
+import SwiftSyntax
+import XCTest
+import _SwiftSyntaxTestSupport
+
+final class FixItTests: XCTestCase {
+  func testEditsForFixIt() throws {
+    let markedSource = "protocol 0️⃣Multi 1️⃣ident 2️⃣{}"
+    let (markers, source) = extractMarkers(markedSource)
+    let positions = ["0️⃣", "1️⃣", "2️⃣"].compactMap { markers[$0] }
```

Comment:
A few nitpicks:
- We start counting at `1` for the location markers in the other test cases, would be nice to be consistent here as well.
- Instead of using the `compactMap` here and then doing `positions[<integer>]` below, I would prefer to use `marker["1️⃣"]` below. Again just to be consistent with the other test cases.

Reply:
compactMap here is to avoid force-unwrapping marker value. i.e. marker["1️⃣"]!, but you're right, marker["1️⃣"] is more elegant. Let me force-unwrap the value just like what we did [here](https://github.com/apple/swift-syntax/blob/06b417950cdbe7f5591bb6849bf36c40fcfe69ad/Tests/SwiftDiagnosticsTest/GroupDiagnosticsFormatterTests.swift#L85).